uni/lib/main.dart

```diff
@@ -120,6 +122,27 @@ Future<void> main() async {
     Logger().w('Plausible is not enabled');
   }
 
+  final appLinks = AppLinks();
+  appLinks.uriLinkStream.listen((uri) async {
+    Logger().d('AppLinks intercepted: $uri');
+    await closeInAppWebView();
+    if (uri.host == 'auth') {
+      final sessionProvider = stateProviders.sessionProvider;
+      try {
+        await sessionProvider.finishFederatedAuthentication(uri);
+        final context = sessionProvider.context;
+        if (context != null && context.mounted) {
+          await Navigator.pushReplacementNamed(
+            context,
+            '/${NavigationItem.navPersonalArea.route}',
+          );
+        }
+      } catch (err) {
+        Logger().e('Failed to login with FederatedLogin: $err');
+      }
+    }
+  });
+
```

Comment:
Maybe the handler function should be abstracted somewhere else, to keep the login code more coupled